grails-app/controllers/org/pih/warehouse/product/ProductController.groovy

```diff
@@ -1014,14 +1014,18 @@ class ProductController {
     def addSynonymToProduct = {
         println "addSynonymToProduct() " + params
         Product product = null
+        def inputValues = null
         try {
             product = productService.addSynonymToProduct(params.id, params.synonymTypeCode, params.synonym, params.locale)
         } catch (IllegalArgumentException e) {
             // If adding a synonym fails, we still want to return the product to the view
             product = Product.read(params.id)
+            // If a validation error occurs, we want to return those values to the view,
+            // and set them as initialValues of the form, so a user can correct him/herself
+            inputValues = params
             flash.error = e.message
         }
-        render(template: 'productSynonyms', model: [productInstance: product])
+        render(template: 'productSynonyms', model: [productInstance: product, inputValues: inputValues])
```

Comment:
If I returned `params`, I would also return them in case if the synonym was successfully added - I want to return them only if an error occurs, that's why I created this additional variable.
Do you have any other idea of handling this?

Reply:
Ok, I guess it makes sense